src/Microsoft.Framework.ConfigurationModel.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.Framework.ConfigurationModel.UserSecrets;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Infrastructure;
+
+namespace Microsoft.Framework.ConfigurationModel
+{
+    public static class ConfigurationExtensions
+    {
+        /// <summary>
+        /// Adds the user secrets configuration source.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public static IConfigurationSourceRoot AddUserSecrets([NotNull]this IConfigurationSourceRoot configuration)
+        {
+            var appEnv = (IApplicationEnvironment)CallContextServiceLocator.Locator.ServiceProvider.GetService(typeof(IApplicationEnvironment));
```

Comment:
Make sure we add a note in https://github.com/aspnet/Configuration/pull/160 so that this doesn't slip through.